### PR TITLE
fix(design): improve design slightly

### PR DESCRIPTION
**Pull request by flyck**
I feel the design is not as nice as it could be. Made some random changes. Looks better? (left: before, right: after)

Changes:
- currently only in white mode the header is transparent, now also in dark mode (has cool effect when reading a blogpost!)
- dark mode header now was a tiny border
- footer has a different color
- there is more of a margin between the highlighted posts on the main page and the footer
- less margins on the codeblocks, right now they seem a bit puffy (🤣)

Screenshot 2023-09-15 at 18 45 32
Screenshot 2023-09-15 at 18 47 14
Screenshot 2023-09-15 at 18 51 09

### CODE REVIEW

File: src/components/blog/HighlightedPosts.astro [hunk 1]
```diff
@@ -22,4 +22,5 @@ const posts = await findPostsByIds(ids);
 		</h2>
 	</div>
 	<Grid posts={posts} />
+        <div class="mb-12" />
```
Comment (hendriknielaender) [inline src/components/blog/HighlightedPosts.astro:25]: Why an extra div? can't you add that `mb-12` to the `<Grid />` component?

File: src/assets/styles/base.css [hunk 2]
```diff
@@ -33,8 +33,8 @@
 
 .astro-code {
   font-size: 0.9rem !important;
-  line-height: 1.25 !important;
-  padding: 0.8rem;
+  line-height: 1.1 !important;
+  padding: 0.4rem;
```
Comment (hendriknielaender) [inline src/assets/styles/base.css:37]: Now it feels like the is no real space, i guess something like 0.6 would be better here.

Comment (flyck) [reply src/assets/styles/base.css:37]: done!

File: src/components/blog/HighlightedPosts.astro (hunk 1 above)
Comment (flyck) [reply src/components/blog/HighlightedPosts.astro]: done!

**Pull request merged**